### PR TITLE
refactor: improve on ResourceCheck functions

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:
- add test file for **parseVariableToName**
- using guard statement in CheckResourceDestroy and CheckResourceExists
- using log and fmt packages

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
$ go test ./huaweicloud/services/acceptance/ -v
=== RUN   TestResourceCheck_parseVariable
--- PASS: TestResourceCheck_parseVariable (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance   0.085s

$ make testacc TEST='./huaweicloud/services/acceptance/cci' TESTARGS='-run TestAccCciNamespace_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cci -v -run TestAccCciNamespace_basic -timeout 360m -parallel 4
=== RUN   TestAccCciNamespace_basic
=== PAUSE TestAccCciNamespace_basic
=== CONT  TestAccCciNamespace_basic
--- PASS: TestAccCciNamespace_basic (47.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cci       47.504s

$ make testacc TEST='./huaweicloud/services/acceptance/css' TESTARGS='-run TestAccCssFlavorsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/css -v -run TestAccCssFlavorsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccCssFlavorsDataSource_basic
=== PAUSE TestAccCssFlavorsDataSource_basic
=== CONT  TestAccCssFlavorsDataSource_basic
--- PASS: TestAccCssFlavorsDataSource_basic (80.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/css       80.856s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/resource_check.go [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+)
+
+/*
+InitDataSourceCheck build a 'ResourceCheck' object. Only used to check datasource attributes.
+  Parameters:
+    resourceName: The resource name is used to check in the terraform.State. e.g. huaweicloud_waf_domain.domain_1
+  Return:
+    *ResourceCheck: ResourceCheck object
+*/
+func InitDataSourceCheck(sourceName string) *ResourceCheck {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/resource_check.go]: The comment name and the argument name are not equal. pls check other places.

File: huaweicloud/services/acceptance/acceptance.go [hunk 2]
```diff
@@ -259,34 +254,36 @@ func (rc *ResourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
...
-					rc.resourceName, rs.Primary.ID, err)
-			}
-			json.Unmarshal(b, rc.resourceObject)
-		} else {
-			logp.Printf("[WARN] The 'resourceObject' is nil, please set it during initialization.")
-		}
+		json.Unmarshal(b, rc.resourceObject)
 	} else {
-		return fmtp.Errorf("The 'getResourceFunc' is nil, please set it.")
+		log.Printf("[WARN] The 'resourceObject' is nil, please set it during initialization")
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/acceptance.go]: suggest move to up, keep same with CheckResourceDestroy

File: huaweicloud/services/acceptance/acceptance.go [hunk 3]
```diff
@@ -225,16 +219,21 @@ func TestCheckResourceAttrWithVariable(resourceName, key, varStr string) resourc
 // CheckResourceDestroy check whether resources destroyed in HuaweiCloud.
 func (rc *ResourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
 	if strings.Compare(rc.resourceType, dataSourceTypeCode) == 0 {
-		logp.Printf("Error, you built a ResourceCheck with 'InitDataSourceCheck', " +
-			"it cannot run CheckResourceDestroy().")
+		log.Printf("[WARN] builting a ResourceCheck with 'InitDataSourceCheck', " +
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/acceptance.go]: unnecessary, just remove this log

File: huaweicloud/services/acceptance/acceptance.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/acceptance.go]: sure, that will be more clear

File: huaweicloud/services/acceptance/acceptance.go (hunk 3 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/acceptance.go]: ok, we can remove it

File: huaweicloud/services/acceptance/resource_check.go (hunk 1 above)
Comment (ShiChangkuo) [reply huaweicloud/services/acceptance/resource_check.go]: fixed

**Pull request merged**